src/app/profiles/profiles.editor.modal.ts

```diff
@@ -0,0 +1,583 @@
+            signature  : data.profile.signature,
+            smtp_address  : data.profile.smtp_address,
+            smtp_port     : data.profile.smtp_port,
+            smtp_username : data.profile.smtp_username,
+            smtp_password : data.profile.smtp_password,
+            type          : this.type,
+            is_smtp_enabled : ( data.profile.is_smtp_enabled ? 1 : 0 ),
+        }
+        let req = this.rmm.profile.create(values, this.field_errors)
+        req.subscribe(
+          data => {
+            let reply = data;
+            if ( reply.status == 'success' ) {
+                this.is_busy = false;
+                this.rmm.profile.load()
+                this.close();
+                return;
+            }
+          },
+        );
+    }
+    delete(){
+        this.is_busy = true;
+        let data = this.data;
+        let req = this.rmm.profile.delete(data.profile.id)
+        req.subscribe(data => {
+            let reply = data
+            if ( reply.status == 'success' ) {
+                this.is_busy = false;
+                this.rmm.profile.load()
+                this.close();
+                return;
+            }
+        });
+    }
+    update(){
+        this.is_busy = true;
+        let data = this.data;
+        let values = {
+            name       : data.profile.name,
+            email      : data.profile.email,
+            from_name  : data.profile.from_name,
+            reply_to   : data.profile.reply_to,
+            signature  : data.profile.signature,
+            smtp_address  : data.profile.smtp_address,
+            smtp_port     : data.profile.smtp_port,
+            smtp_username : data.profile.smtp_username,
+            smtp_password : data.profile.smtp_password,
+            is_smtp_enabled : ( data.profile.is_smtp_enabled ? 1 : 0 ),
+        }
+        let req = this.rmm.profile.update(this.data.profile.id, values, this.field_errors)
+        req.subscribe(
+          data => {
+            let reply = data;
+            this.is_busy = false;
+            if ( reply.status == 'success' ) {
+                this.rmm.profile.load()
+                this.close();
+                return;
+            } else {
+                console.log('ERROR:', reply)
+                if ( reply.field_errors ) {
+                    this.field_errors = reply.field_errors;
+                }
+            }
+          },
+        );
+    }
+    close() {
+        this.dialog_ref.close({});
+    }
+    show_error (message, action) {
+      this.snackBar.open(message, action, {
+        duration: 2000,
+      });
+    };
+    onchange_field ( field ) {
+        if ( this.field_errors && this.field_errors[field] ) {
+            this.field_errors[field] = [];
+        }
+        if ( field === 'preferred_runbox_domain' ) {
+            const data = this.data;
+            const selected_domain = data.profile.preferred_runbox_domain;
+            ['email'].forEach((attr)=>{
```

Comment:
Is there a need for this `forEach` here if it's just a single string?

Reply:
because it had the 'reply_to' field